ground/src/main/java/com/google/android/ground/persistence/remote/firebase/protobuf/ModelToProtoExt.kt

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.persistence.remote.firebase.protobuf
+
+import com.google.android.ground.model.User
+import com.google.android.ground.model.geometry.Coordinates
+import com.google.android.ground.model.geometry.Geometry
+import com.google.android.ground.model.geometry.LineString
+import com.google.android.ground.model.geometry.LinearRing
+import com.google.android.ground.model.geometry.MultiPolygon
+import com.google.android.ground.model.geometry.Point
+import com.google.android.ground.model.geometry.Polygon
+import com.google.android.ground.model.locationofinterest.LoiProperties
+import com.google.android.ground.model.mutation.LocationOfInterestMutation
+import com.google.android.ground.model.mutation.Mutation
+import com.google.android.ground.proto.AuditInfo as AuditInfoProto
+import com.google.android.ground.proto.Coordinates as CoordinatesProto
+import com.google.android.ground.proto.LinearRing as LinearRingProto
+import com.google.android.ground.proto.LocationOfInterest as LocationOfInterestProto
+import com.google.android.ground.proto.LocationOfInterest.Property as PropertyProto
+import com.google.android.ground.proto.LocationOfInterest.Source
+import com.google.android.ground.proto.LocationOfInterestKt.property
+import com.google.android.ground.proto.MultiPolygon as MultiPolygonProto
+import com.google.android.ground.proto.Point as PointProto
+import com.google.android.ground.proto.Polygon as PolygonProto
+import com.google.android.ground.proto.coordinates
+import com.google.android.ground.proto.geometry
+import com.google.android.ground.proto.point
+import com.google.protobuf.timestamp
+import java.util.Date
+
+fun LocationOfInterestMutation.createLoiMessage(user: User): LocationOfInterestProto {
+  assert(userId == user.id) { "UserId doesn't match: expected $userId, found ${user.id}" }
+
+  val builder =
+    LocationOfInterestProto.newBuilder()
+      .setId(locationOfInterestId)
+      .setJobId(jobId)
+      .setSubmissionCount(submissionCount)
+      .setOwnerId(userId)
+      .setCustomTag(customId)
+
+  if (properties.isNotEmpty()) {
```

Comment:
The `{ }` blocks in the DSL are just Kotlin lambdas, so you should be able to conditionally set values as well.

Reply:
Also, I think an empty map is the default proto map field value when unset anyway, so you could just set it in any case.